src/components/UserSelector.tsx

```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type UserSelectorProps = {
+  users: User[];
+  setSelectedUser: (user: User) => void;
+  selectedUser: User | null;
+};
+
+export const UserSelector = ({
+  users,
+  setSelectedUser,
+  selectedUser,
+}: UserSelectorProps) => {
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+
+  const handleClick = (user: User) => {
+    setSelectedUser(user);
+    setDropdownOpen(false);
+  };
+
+  const handleClickOutside = () => {
+    if (!dropdownOpen) {
+      setDropdownOpen(false);
+    }
+  };
+
+  useEffect(() => {
+    document.body.addEventListener('mousedown', handleClickOutside);
```

Comment:
Consider to add cleanup function to remove the event listener when the component is unmounted.

`return () => window.body.removeEventListener(`